### PR TITLE
[veomni, fsdp] feat: enable fused top-K distillation kernel for OPD

### PR DESCRIPTION
**Pull request by Luosuu**
### What does this PR do?

Wires veomni's `chunk_topk_distill_function` into the verl actor
pipeline so on-policy distillation (OPD) batches can run with
`use_fused_kernels=True`. The fused kernel computes the top-K
forward-KL distillation loss without materializing the full
`[B, L, V]` logits tensor, saving significant GPU memory for
large-vocabulary models.

Previously, distillation TDs had to force `use_fused_kernels=False`
because the fused path only produced per-token log_probs/entropy —
the in-forward logit processor needed full logits to compute the
top-K weighted CE. With veomni 0.1.11's `chunk_topk_distill_function`,
the distillation loss is computed inline during the chunked
fused-linear forward pass.

### Design & Code Changes

Engine changes (two files, additive — no existing behavior modified):

- **`VeOmniEngineWithLMHead.prepare_model_inputs`**: when
  `use_fused_kernels=True`, `distillation_use_topk=True`, and
  `teacher_topk_ids` exists in the micro-batch, flattens the nested
  teacher tensors to `(1, total_nnz, K)` and passes them into
  `model_inputs`. They propagate as `**kwargs` through the model
  forward to `ForCausalLMLoss`, which routes to
  `chunk_topk_distill_function`. Includes paired validation (raises
  `ValueError` if `teacher_topk_ids` present without
  `teacher_topk_log_probs`). Guards on key existence so verl's
  native distillation path (different tensor keys) is unaffected.

- **`FSDPEngineWithLMHead.prepare_model_outputs`**: in the
  `use_fused_kernels=True` block, extracts `distillation_losses` /
  `student_mass` / `teacher_mass` from `output.fused_linear_aux` and
  stores them as nested tensors in `model_output` — same keys and
  format as the eager logit-processor path, so downstream loss
  functions work unchanged.

Both paths (fused and eager) remain fully functional:
- **Fused** (`use_fused_kernels=True`, VeOmni engine): `ForCausalLMLoss`
  routes to `chunk_topk_distill_function`.
- **Eager** (`use_fused_kernels=False`, FSDP/Megatron): existing
  in-forward logit processor fires as before.

CI: bump veomni from 0.1.10 to 0.1.11 (adds the `chunk_topk_distill`
kernel).

### Test

- **2xH100 unit test with Qwen3-0.6B**: synthetic distillation TD with
  `use_fused_kernels=True` and `distillation_use_topk=True`. Verified
  `distillation_losses`, `student_mass`, `teacher_mass` present as
  nested tensors in `model_output`, all values finite and non-negative,
  both ranks produce consistent output. Confirms our new engine code
  paths (input injection, output extraction) work correctly.

- **Full e2e example** (`run_qwen3_8b_mopd_veomni.sh` /
  `run_qwen3_0.6b_opd_veomni.sh` with verl native distillation):
  reaches actor backward on H100x8 then hits a pre-existing veomni
  `chunk_logprobs` + FSDP2 bug — see
  https://github.com/ByteDance-Seed/VeOmni/issues/804. The bug is in
  veomni's existing fused log_probs kernel (not added by this PR) and
  fires whenever `use_fused_kernels=True` is set for any VeOmni FSDP2
  actor with a backward pass, regardless of distillation. Our engine
  changes are downstream of this bug — verified path-clean by the
  unit test above. E2E example will run end-to-end once veomni #804
  lands.

### Dependencies

Requires veomni >= 0.1.11 for `chunk_topk_distill_function` and
`FusedLinearAuxOutput.distillation_losses`.

### CODE REVIEW

File: verl/workers/engine/fsdp/transformer_impl.py [hunk 1]
```diff
@@ -1092,6 +1092,21 @@ def prepare_model_outputs(self, output, output_args, micro_batch: TensorDict, lo
                 # temperature is singleton
                 log_probs = output.log_probs.squeeze(0)  # (total_nnz,)
                 entropy_rmpad = output.entropy.squeeze(0)  # (total_nnz,)
+
+                # When the fused kernel also computed top-K distillation
+                # (veomni's chunk_topk_distill path), extract the per-token
+                # distillation outputs and store them as nested tensors —
+                # same model_output keys as the eager logit-processor path.
+                if distillation_use_topk:
```
Comment (wuxibin89) [inline verl/workers/engine/fsdp/transformer_impl.py:1100]: Does veomni `chunk_topk_distill_function` also apply to fsdp backend?

Comment (Luosuu) [reply verl/workers/engine/fsdp/transformer_impl.py:1100]: Not yet.

**Review by wuxibin89 — APPROVED** [review]
